### ISSUE TITLE
Migrate existing data

### ISSUE THREAD

**Issue by matthijskooijman** [evolution-events/Artaxerxes#48]
We probably need to migrate existing data, for historical purposes.

One challenge is migrating users. Should we migrate everyone including their passwords? Or without and let them use "forgot my password"? What if they want to convert to a social login right away? That would create a new account (possibly producing a conflict on the mail-address), rather than being merged with the existing account? Perhaps a custom migration link that we can send to users?

**Comment by ReneB** [comment]
Most data does not need to be migrated _before_ our first registrations, I guess. However, we may want to  migrate users already.

The password was very PHPBB-specific, I seem to remember, and might not be worth the effort of implementing it for ourselves in this system. "Recovery" - as in, let everyone choose a new password - would be the easier way, I guess.

With respect to signing up through a social account immediately, I could imagine an admin interface to link local accounts to social users, which should then also allow linking newly made social accounts to imported users.

**Comment by matthijskooijman** [comment]
> Most data does not need to be migrated before our first registrations, I guess. However, we may want to migrate users already.

Yeah, we should definitely migrate users, otherwise it will be a mess to untangle later.

> The password was very PHPBB-specific, I seem to remember, and might not be worth the effort of implementing it for ourselves in this system. "Recovery" - as in, let everyone choose a new password - would be the easier way, I guess.

If the PHPBB-password-hash is easy to implement, maybe we should, but just forcing everyone through recovery is also fine.

> With respect to signing up through a social account immediately, I could imagine an admin interface to link local accounts to social users, which should then also allow linking newly made social accounts to imported users. This feature, however, should not be required before the first registrations open.

IIRC the user registrations django app we use already offers an account management page where you can connect auth methods to your account, so this is mostly a matter of embedding this and making it clear to our users how it works (see #53).

**Comment by ReneB** [comment]
Ensure we don't import data into accounts where the email address has not been verified, _or_ we ensure that you cannot access the list of old events before your mail address is verified. The reason for this is that otherwise I could register an account for henk@gmail.com, log in, see all the relevant data about older events, without actually owning henk@gmail.com.

The _real_ solution to this is fixing #87.
